### PR TITLE
chore: Prepare 0.12.0 release

### PR DESCRIPTION
**Pull request by binarylogic**
Release data and note for 0.12.0.

### CODE REVIEW

**Comment by binarylogic** [comment]
@timberio/vector could everyone review the upgrade guide and make sure we are not missing any breaking changes or deprecations that you worked on? Thank you.

**Review by ktff — APPROVED** [review]
:tada:

File: docs/highlights/2021-01-31-file-source-checkpointing.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+author_github: "https://github.com/binarylogic"
+featured: false
+pr_numbers: [6178]
+release: "0.12.0"
+hide_on_release_notes: false
+tags: ["domain: sources", "source: file"]
+---
+
+The Vector [`file` source][file_source] included an option called `start_at_beginning` that would determine where
+Vector would begin reading a file based on a [variety of conditions]. As you can see, these conditions were quite
```
Comment (lucperkins) [inline docs/highlights/2021-01-31-file-source-checkpointing.md:15]: The URL isn't specified for this link

File: docs/reference/configuration.cue [hunk 2]
```diff
@@ -133,5 +133,34 @@ configuration: {
 				```
 				"""
 		}
+		wildcards: {
+			title: "Wildcards in identifiers"
+			body:  """
+				Vector supports wildcards (`*`) in component identifiers when building your topology, but only supports
+				it as the last character. For example:
```
Comment (lucperkins) [inline docs/reference/configuration.cue:140]: s/it/them

File: docs/reference/releases/0.12.0.cue [hunk 3]
```diff
@@ -0,0 +1,211 @@
+package metadata
+
+releases: "0.12.0": {
+	date:     "2021-02-11"
+	codename: "Vector Remap Language"
+	description: """
+		The Vector team is pleased to announce 0.12.0. This release	introduces the **Vector Remap Language**, an
```
Comment (lucperkins) [inline docs/reference/releases/0.12.0.cue:7]: There are a few accidental tabs on this page. Not sure if that matters in the presented text.

**Review by lucperkins — APPROVED** [review]

File: docs/highlights/2021-02-11-0-12-upgrade-guide.md [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+-lanes.errors."level.eq" = "error"
++lanes.errors = '.level = "error"'
+```
+
+Refer to the [VRL reference][vrl_reference] for the many ways you can specify conditions.
+
+### The `generator` source requires a `format` option
+
+The [`generator` source], commonly used for testing, has been updated with a new `format` option that emits logs in
+the specified format. You will not be required to provide this option. Upgrading is easy:
```
Comment (jszwedko) [inline docs/highlights/2021-02-11-0-12-upgrade-guide.md:81]: Typo, I think this should say `now`.

**Pull request merged**